[Mobile Payments] Clean up feature flags

joshheald opened the pull request:
<!-- Remember about a good descriptive title. -->

<!-- Id number of the GitHub issue this PR addresses. -->

## Description
<!-- Take the time to write a good summary. Why is it needed? What does it do? When fixing bugs try to avoid just writing “See original issue” – clarify what the problem was and how you’ve fixed it. -->

This PR removes a number of payments-related feature flags, for changes which were previously released.

It's not urgent, so I've tagged it for 16.2.

I've split the removals up into individual commits, which may be an easier way to review this.

## Testing instructions
<!-- Step by step testing instructions. When necessary break out individual scenarios that need testing, consider including a checklist for the reviewer to go through. -->

Check that unit tests pass.

Check the following features work as expected:

- [ ] For a new install of the app, when completing IPP onboarding you're prompted to enable Pay in Person
- [ ] For a Woo Express free trial site with a US app store account, In-App Purchases are offered for upgrades
- [ ] Just in Time Messages are shown on the dashboard (test without the throttle, you'll need a sandbox)
- [ ] IPP feedback banner is shown on the Order List
- [ ] Retrying a failed IPP works
- [ ] Discounts can be added to a product in an order, and the UX is as expected
- [ ] In person payments for orders work, and a reload of the order is done when the payment is started
- [ ] Manual taxes work as expected
- [ ] Do a release build to a simulator, check in `Menu > Settings > Experimental Features` – an `In App Purchases` toggle should **not** be visible.


---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

joshheald committed:
Remove Pay in Person feature flag

joshheald committed:
Remove unused inAppPurchases feature flag

joshheald committed:
Remove JITM on dashboard feature flag

joshheald committed:
Remove IPP Feedback banner feature flag

joshheald committed:
Remove retry payment with reused PI feature flag

joshheald committed:
Remove ordersWithCoupons feature flags

joshheald committed:
Remove refreshOrderBeforeIPP feature flag

joshheald committed:
Tidy up unused feature flags

joshheald committed:
Remove manualTaxesInOrderM2 feature flag

joshheald committed:
Remove manualTaxesInOrderM3 feature flag

joshheald committed:
Fix whitespace warning

joshheald committed:
Merge branch 'trunk' into task/clean-up-feature-flags

joshheald commented:
Making notes of my testing here:
- [x] For a new install of the app, when completing IPP onboarding you're prompted to enable Pay in Person
- [x] For a Woo Express free trial site with a US app store account, In-App Purchases are offered for upgrades
- [x] Just in Time Messages are shown on the dashboard (test without the throttle, you'll need a sandbox)
- [x] IPP feedback banner is shown on the Order List
- [x] Retrying a failed IPP works
- [x] Discounts can be added to a product in an order, and the UX is as expected
- [x] In person payments for orders work, and a reload of the order is done when the payment is started
- [x] Manual taxes work as expected
- [x] Do a release build to a simulator, check in `Menu > Settings > Experimental Features` – an `In App Purchases` toggle should **not** be visible.

joshheald commented:
@iamgabrielma @toupper Tagging you both because this clears down feature flags that we've all worked on. Only needs one review, but a quick look over from each of you would be appreciated in case you've particular concerns about removing any of these flags.

Please post on here when you've checked; it's tagged for 16.2 so no rush to do it this week.

Thanks!

joshheald committed:
Merge branch 'trunk' into task/clean-up-feature-flags

toupper commented on Experiments/Experiments/DefaultFeatureFlagService.swift:
```diff
@@ -22,16 +22,8 @@ public struct DefaultFeatureFlagService: FeatureFlagService {
         case .searchProductsBySKU:
             return true
-        case .inAppPurchases:
-            return buildConfig == .localDeveloper || buildConfig == .alpha
```
As I see, this wasn't part of the release, but now, when removing the feature flag, it will be. Is this the intended behavior? Do we want release notes for this change?

joshheald replied:
Argh thank you for spotting this. I thought it wasn't used, but actually it is, for the debug IAP menu. We don't really want the toggle to turn that debug menu on starting to show up in production builds.

I've put the flag back in 7dc6b4a, renamed it for clarity, and I'll let Paolo know that we've left it in place. They may just want to remove that menu, or at least make it permanently only available in debug builds.

joshheald committed:
Revert "Remove unused inAppPurchases feature flag"

This reverts commit fae9ec236d5b53375364d224adcd897dc01101e5.

# Conflicts:
#	Experiments/Experiments/DefaultFeatureFlagService.swift

joshheald committed:
Rename IAP feature flag for clarity

This was mistakenly enabled during a cleanup, which would have made the debug menu available in production builds. To avoid that in future, I’ve renamed and clarified the purpose of this flag.

toupper reviewed (approved):
Thanks for taking care of this! Removing code is always good 👏 🚢
